db/memtable_list.cc

```diff
@@ -609,6 +668,8 @@ Status InstallMemtableAtomicFlushResults(
                          cfds[i]->GetName().c_str(), m->GetFileNumber(),
                          mem_id);
         imm->current_->Remove(m, to_delete);
+        imm->UpdateMemoryUsageExcludingLast();
+        imm->ResetTrimHistoryNeeded();
```

Comment:
This is after atomic flush is committed to MANIFEST. Do we always reset this flag at this point? What happens if we already have sufficient memtables history and the `trim_history_needed` is true. Then a flush is triggered and completed. Is it correct to reset the flag in this case?

Reply:
please refer to this comment above:  https://github.com/facebook/rocksdb/pull/5022#discussion_r307534169